tests/core/server/test_dos.py

```diff
@@ -201,9 +201,15 @@ async def test_spam_tx(self, setup_two_nodes_fixture, self_hostname):
         # Remove outbound rate limiter to test inbound limits
         ws_con.outbound_rate_limiter = RateLimiter(incoming=True, percentage_of_limit=10000)
 
-        for i in range(6000):
-            await ws_con._send_message(new_tx_message)
-        await asyncio.sleep(1)
+        try:
+            for i in range(6000):
+                await ws_con._send_message(new_tx_message)
+            await asyncio.sleep(1)
+        except ConnectionResetError:
+            # based on timing the connection may get closed by rate limits during the loop
+            # and this is actually want we want so catch this exception and ignore it
+            # seen primarily on my Windows box
+            pass
```

Comment:
Typo in `want we want`.  The sleep could go outside as well, I think?

Why do we not get the exception everywhere else?  This seems like a good thing to understand a bit better.

Reply:
The rate limits are x messages within x timeframe - we actually want it to fire sometime here in the loop. I kinda would have expected this exception in more places actually.

Reply:
The connection does close, that is checked in line 217. So it must be one of two scenarios:
1. On other platforms, there is a delay when closing connection
2. On other platforms, you can still call ws_con.send_message on a closed WS.

I'm not sure which is the right one

Reply:
https://github.com/Chia-Network/chia-blockchain/pull/11612

Reply:
I was thinking of the tests could be rewritten a little bit - we can use the code in rate_limits.py to use the exact number of messages that should send it over the limit - rather than this rather obscure 6000. I believe the frequency limit for NewTransaction is 5000. This is also testing the inbound limit, so the other end needs to close the connection - I suspect Windows ends up closing the connection much faster than Unix - so my guess is it's (1). That said, I'm not sure how much it matters to track it down exactly.

Reply:
I think with kyle's fix we are good, there is no need to spend more time on this

Reply:
the tests are flaky anyways.  unless mariano ended up handling this.  i had started working on that in https://github.com/Chia-Network/chia-blockchain/pull/9206 a long time ago.  but that's just general, not about windows.

Reply:
Undoing changes here to allow https://github.com/Chia-Network/chia-blockchain/pull/11612 to fix it when it gets merged.

```suggestion
        for i in range(6000):
            await ws_con._send_message(new_tx_message)
        await asyncio.sleep(1)
```